cmd/out/main.go

```diff
@@ -117,6 +118,26 @@ func main() {
 
 	logrus.Info("pushed")
 
+	if req.Source.ContentTrust.Enable {
+		notaryConfigDir, err := req.Source.ContentTrust.PrepareConfigDir(src)
+		if err != nil {
+			logrus.Errorf("failed to prepare notary-config-dir: %s", err)
+			os.Exit(1)
+			return
+		}
+		os.Setenv("NOTARY_CONFIG_DIR", notaryConfigDir)
```

Comment:
This seems kinda odd; could it be passed as an argument instead? :thinking:

Reply:
Since `github.com/theupdateframework/notary` rely on a `trust_directory`. In order to keep consistency, I'm using `NOTARY_CONFIG_DIR` to explicitly specified the directory.

Reply:
I just meant that it felt odd to have to set an env var in order to use a Go API. It's kind of a mix of system-level config and language-level config. Could this be pushed under the hood and handled by your library? Where does this env var get used? I can't seem to find it anywhere.

Reply:
It looks like it gets parsed out in [`gcr.NewTrustedGcrRepository`](https://github.com/simonshyu/notary-gcr/blob/8d81a216e1287b6e8d51078c3b07da0ecfb1d7e7/pkg/gcr/gcr.go#L18-L25). The config gets written to disk and then loaded up immediately after within the same Go process, which is what feels weird.

Does something else use this directory format? If not, couldn't the config just be passed directly to the API?

Does anything else use that same env var? If not, couldn't it just be passed as an argument?

Thanks!

Reply:
OK, I will passed it as an argument.